test/xbgpu/test_engine.py

```diff
@@ -470,7 +564,7 @@ def sensor_observer(sync_sensor: aiokatcp.Sensor, sensor_reading: aiokatcp.Readi
             """Record sensor updates in a list for later comparison."""
             actual_sensor_updates.append((sensor_reading.value, sensor_reading.status))
 
-        xbengine.sensors["rx.synchronised"].attach(sensor_observer)
+        xbengine.sensors[f"{corrprod_outputs[0].name}.rx.synchronised"].attach(sensor_observer)
```

Comment:
Check it for every corrprod, not just the first.

Reply:
I didn't expect this to be a quick fix, but I also didn't expect to run into the issue I have. Decided to update this to fill a dictionary (key == sensor name) and the value is a list of tuples.
* Each update is seemingly called once (correctly) - verified with logging statements,
* But the eventual dictionary shows updates from both sensors present in both lists.

Same issue when generating the `expected_sensor_updates` later on. Will keep plugging at it, but just had to share this strangeness.

Reply:
Yeah I figured I hadn't discovered a new branch of quantum entanglement. Thank you for the reminder.